### PR TITLE
Aiv 2270

### PR DESCRIPTION
**Pull request by YuanZHANG910**

### CODE REVIEW

File: app/engine/AuditInfo.scala [hunk 1]
```diff
@@ -17,34 +17,26 @@
...
-
 
 sealed trait TAuditInfo {
   lazy val transactionNames = Map(
     Locations.PersonalTaxAccount -> "sent to personal tax account",
     Locations.BusinessTaxAccount -> "sent to business tax account"
   )
 
-  def toAuditEvent(location: Location)(implicit hc: HeaderCarrier, authContext: AuthContext, request: Request[AnyContent]): ExtendedDataEvent = {
+  def toAuditEvent(location: Location)(implicit hc: HeaderCarrier, authContext: Enrolments, request: Request[AnyContent]): ExtendedDataEvent = {
```
Comment (adwelly) [inline app/engine/AuditInfo.scala:35]: should the param authContext be renamed to enrolments?

File: app/engine/AuditInfo.scala [hunk 2]
```diff
@@ -17,34 +17,26 @@
...
     this match {
       case AuditInfo(routingReasons, ruleApplied, throttlingInfo) =>
-        val accounts = authContext.principal.accounts
-        val accountMap = accounts.toMap
-        val accountsAsJson: Seq[(String, JsValueWrapper)] = accountMap
-          .map { case (k, v) => (k, Json.toJsFieldJsValueWrapper(v.toString)) }
-          .toSeq
-        val optionalAccounts: JsObject = Json.obj(accountsAsJson: _*)
+        val enrolments: Set[Enrolment] = authContext.enrolments
```
Comment (adwelly) [inline app/engine/AuditInfo.scala:38]: Hmm. perhaps not?

File: app/engine/AuditInfo.scala (hunk 1 above)
Comment (YuanZHANG910) [reply app/engine/AuditInfo.scala:35]: will do

File: app/controllers/RouterController.scala [hunk 3]
```diff
@@ -88,12 +73,12 @@ trait RouterController extends FrontendController with Actions {
   }
 
   def sendAuditEvent(ruleContext: RuleContext, auditInfo: AuditInfo, throttledLocation: Location)
-                    (implicit authContext: AuthContext, request: Request[AnyContent]) = {
+                    (implicit authContext: Enrolments, request: Request[AnyContent]): Unit = {
```
Comment (adwelly) [inline app/controllers/RouterController.scala:76]: Naming issue again?

File: app/model/RuleContext.scala [hunk 4]
```diff
@@ -17,61 +17,73 @@
...
-  val authConnector: FrontendAuthConnector = FrontendAuthConnector
-  val userDetailsConnector: UserDetailsConnector = UserDetailsConnector
-
-  lazy val userDetails = authority.flatMap { authority =>
-    authority.userDetailsUri.map(userDetailsConnector.getUserDetails).getOrElse {
-      logger.warn("failed to get user details because userDetailsUri is not defined")
-      Future.failed(new RuntimeException("userDetailsUri is not defined"))
-    }
+  def isVerifyUser(implicit hc: HeaderCarrier): Future[Boolean] = {
+    authorised(AuthProviders(AuthProvider.Verify)){
```
Comment (adwelly) [inline app/model/RuleContext.scala:36]: Will this force a login if they are not authorised? or does it merely check to see what the situation is?

File: it/router/RouterAnalyticsFeature.scala [hunk 5]
```diff
@@ -16,32 +12,22 @@ class RouterAnalyticsFeature extends StubbedFeatureSpec with CommonStubs {
...
-      And("a user logged in through Government Gateway")
-      val saUtr = "12345"
-      val accounts = Accounts(sa = Some(SaAccount("", SaUtr(saUtr))))
-      SessionUser(accounts = accounts, isRegisteredFor2SV = false).stubLoggedIn()
-      stubUserDetails()
+      And("a user logged in through Verify")
 
       And("the user has self assessment enrolments")
-      stubSelfAssessmentEnrolments()
+      setVerifyUser()
```
Comment (adwelly) [inline it/router/RouterAnalyticsFeature.scala:18]: stubSelfAssessmentEnrolments() has gone from here?

File: it/router/RouterAnalyticsFeature.scala [hunk 6]
```diff
@@ -16,32 +12,22 @@ class RouterAnalyticsFeature extends StubbedFeatureSpec with CommonStubs {
...
 
-      createStubs(BtaHomeStubPage)
+      createStubs(PtaHomeStubPage)
 
       When("the user hits the router")
       go(RouterRootPath)
 
-      Then("the user should be routed to BTA home page")
-      on(BtaHomePage)
+      Then("the user should be routed to PTA home page")
```
Comment (adwelly) [inline it/router/RouterAnalyticsFeature.scala:28]: This used to route people to the BTA? Why has it changed?

File: it/router/RouterAnalyticsFeature.scala [hunk 7]
```diff
@@ -16,32 +12,22 @@ class RouterAnalyticsFeature extends StubbedFeatureSpec with CommonStubs {
...
+      on(PtaHomePage)
 
-      And("analytic details were sent to google")
```
Comment (adwelly) [inline it/router/RouterAnalyticsFeature.scala:39]: Do we no longer send analytics?

File: it/router/RouterAuditFeature.scala [hunk 8]
```diff
@@ -604,84 +579,4 @@ class RouterAuditFeature extends StubbedFeatureSpec with CommonStubs with AuditT
   }
 }
 
-class RouterAuditFeatureWithThrottling extends StubbedFeatureSpec with CommonStubs with AuditTools {
```
Comment (adwelly) [inline it/router/RouterAuditFeature.scala:607]: Has this test been entirely deleted?

File: it/router/RouterAnalyticsFeature.scala (hunk 7 above)
Comment (YuanZHANG910) [reply it/router/RouterAnalyticsFeature.scala:39]: I will add this back

**Pull request merged**